WordPress/Classes/Services/EditorSettings.swift

```diff
@@ -67,4 +84,42 @@ class EditorSettings: NSObject {
             database.set(newValue, forKey: nativeEditorEnabledKey)
         }
     }
+
+    // We can't return a type that's both a PostEditor and a UIViewController, so using
+    // a configure block as a hack.
+    // In Swift 4, we'll be able to do `instantiateEditor() -> UIViewController & PostEditor`,
+    // and then let the caller configure the editor.
+    func instantiatePostEditor(post: AbstractPost, configure: (PostEditor, UIViewController) -> Void) -> UIViewController {
+        switch editor {
+        case .aztec:
+            let vc = AztecPostViewController(post: post)
+            configure(vc, vc)
```

Comment:
That's the deceptively tricky bit ;)

To do that, you'd extract the `configure`/`return` out of each individual `case` to after the `switch`. But that means you then have to declare `vc` *before* the `switch`, and what's it's type? In Swift 4, you could do:

```swift
let vc: PostEditor & UIViewController // 👈 Swift 4 feature
switch editor {
case .aztec:
    vc = AztecPostViewController(post: post)
//...
}
configure(vc, vc)
return vc
```

But then you wouldn't need the configure hack 😁